packages/constants/src/lib/errors.ts

```diff
@@ -171,3 +190,103 @@ export const LIT_ERROR = {
 export const LIT_ERROR_CODE = {
   NODE_NOT_AUTHORIZED: 'NodeNotAuthorized',
 };
+
+export abstract class LitError extends VError {
+  protected constructor(
+    options: Error | Options,
+    message: string,
+    ...params: any[]
+  ) {
+    super(options, message, ...params);
+  }
+}
+
+type LitErrorConstructor = new (
+  options: Error | Options,
+  message: string,
+  ...params: any[]
+) => LitError;
+
+function createErrorClass({
+  name,
+  code,
+  kind,
+}: {
+  name: string;
+  code: string;
+  kind: string;
+}): LitErrorConstructor {
+  return class extends LitError {
+    // VError has optional options parameter, but we make it required so thrower remembers to pass all the useful info
+    constructor(options: Error | Options, message: string, ...params: any[]) {
+      if (options instanceof Error) {
+        options = {
+          cause: options,
+        };
+      }
+
+      // If the cause is not an Error, wrap it in one
+      if (!(options.cause instanceof Error)) {
+        options.cause = new Error(options.cause);
+      }
+
+      super(
+        {
+          name,
+          ...options,
+          info: {
+            code,
+            kind,
+            ...options.info,
+          },
+        },
+        message,
+        ...params
+      );
+    }
+  };
+}
+
+const errorClasses: Record<string, LitErrorConstructor> = {};
+for (const key in LIT_ERROR) {
+  if (key in LIT_ERROR) {
+    const errorDef = LIT_ERROR[key];
+    errorClasses[errorDef.name] = createErrorClass(errorDef);
+  }
+}
+
+export const {
```

Comment:
Do we still need to export explicitly snice we are creating an aggregate mapping of `errorClasses` or are we doing named exports to be explicit?

Reply:
I prefer being explicit as when having one file per error; but with so many errors, it's hard to keep track of them all and modify them.
This felt like a good compromise to retain errors individually but centralize their properties and differences based on `ErrorConfig` and `LIT_ERROR`

Reply:
This makes sense to me. Nice inclusion.